linux/LinuxMachine.c

```diff
@@ -603,6 +603,48 @@ static void scanCPUFrequencyFromCPUinfo(LinuxMachine* this) {
    }
 }
 
+#ifdef HAVE_SENSORS_SENSORS_H
+static void LinuxMachine_fetchCPUTopologyFromCPUinfo(LinuxMachine* this) {
+   const Machine* super = &this->super;
+
+   FILE* file = fopen(PROCCPUINFOFILE, "r");
+   if (file == NULL)
+      return;
+
+   int cpuid = -1;
+   int coreid = -1;
+   int physicalid = -1;
+
+   while (!feof(file)) {
+      char buffer[PROC_LINE_LENGTH];
+
+      if (fgets(buffer, PROC_LINE_LENGTH, file) == NULL)
+         break;
```

Comment:
Beware of excessively long lines; do proper EOL handling.

Reply:
All code in that file does it that way, 4096 should be plenty?  Also we only care about short lines.

Reply:
There'S `String_readline` in `XUtils.h` that's fully capable of very long lines and does proper re-assembly.